conn.go

```diff
@@ -182,14 +196,29 @@ func (conn *Conn) Close() error {
 		conn.outLck.Unlock()
 		return nil
 	}
+
 	close(conn.out)
 	conn.closed = true
 	conn.outLck.Unlock()
+
+	close(conn.shutdownSigWorker)
+	conn.signalQueueLck.Lock()
+	// This is to avoid unnecessary wakeup of sigWorker
+	tmp := conn.signalQueueWakeUp
```

Comment:
I don't actually know how this behaves. Is this copying a reference, or is this actually creating a copy of the channel? I would assume the former, but I can't find any documentation that explains this.